cmd/bastion/aws/create.go

```diff
@@ -0,0 +1,549 @@
+		return instanceID, nil
+	}
+
+	// find public subnet
+	subnetID, err := existingSubnet(ctx, ec2Client, infraID)
+	if err != nil {
+		return "", fmt.Errorf("cannot lookup existing subnet: %w", err)
+	}
+	if len(subnetID) == 0 {
+		return "", fmt.Errorf("no public subnet was found")
+	}
+
+	runCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+	result, err := ec2Client.RunInstancesWithContext(runCtx, &ec2.RunInstancesInput{
+		ImageId:      aws.String("resolve:ssm:/aws/service/ami-amazon-linux-latest/amzn2-ami-hvm-x86_64-gp2"),
+		MaxCount:     aws.Int64(1),
+		MinCount:     aws.Int64(1),
+		InstanceType: aws.String("t2.micro"),
+		KeyName:      aws.String(keyPairName(infraID)),
+		NetworkInterfaces: []*ec2.InstanceNetworkInterfaceSpecification{
+			{
+				DeviceIndex:              aws.Int64(0),
+				AssociatePublicIpAddress: aws.Bool(true),
+				SubnetId:                 aws.String(subnetID),
+				Groups:                   []*string{aws.String(sgID)},
+			},
+		},
+		TagSpecifications: []*ec2.TagSpecification{
+			{
+				ResourceType: aws.String("instance"),
+				Tags: []*ec2.Tag{
+					{
+						Key:   aws.String(fmt.Sprintf("kubernetes.io/cluster/%s", infraID)),
+						Value: aws.String("owned"),
+					},
+					{
+						Key:   aws.String("Name"),
+						Value: aws.String(instanceName(infraID)),
+					},
+				},
+			},
+		},
+	})
+	if err != nil {
+		return "", fmt.Errorf("failed to launch bastion instance: %w", err)
+	}
+	for _, instance := range result.Instances {
+		instanceID := aws.StringValue(instance.InstanceId)
+		log.Info("Created ec2 instance", "id", instanceID, "name", instanceName(infraID))
+		return instanceID, nil
+	}
+	return "", fmt.Errorf("no instances were created")
+}
+
+func existingSubnet(ctx context.Context, ec2Client *ec2.EC2, infraID string) (string, error) {
+	var subnetID string
+	subCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+	subnetFilter := []*ec2.Filter{
+		{
+			Name:   aws.String(fmt.Sprintf("tag:kubernetes.io/cluster/%s", infraID)),
+			Values: []*string{aws.String("owned")},
+		},
+	}
+	result, err := ec2Client.DescribeSubnetsWithContext(subCtx, &ec2.DescribeSubnetsInput{
+		Filters: subnetFilter,
+	})
+	if err != nil {
+		return "", fmt.Errorf("cannot list subnets: %w", err)
+	}
+	nameRe := regexp.MustCompile(fmt.Sprintf("%s-public-[a-z,0-9,-]+", infraID))
+	for _, subnet := range result.Subnets {
+		var name string
+		for _, tag := range subnet.Tags {
+			if aws.StringValue(tag.Key) == "Name" {
+				name = aws.StringValue(tag.Value)
+				break
+			}
+		}
+		if len(name) == 0 {
+			continue
+		}
+		if nameRe.MatchString(name) {
+			subnetID = aws.StringValue(subnet.SubnetId)
+			break
+		}
+	}
+	return subnetID, nil
+}
+
+func existingInstance(ctx context.Context, ec2Client *ec2.EC2, infraID string) (string, error) {
+	instanceCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+	instanceFilter := []*ec2.Filter{
```

Comment:
I think I am missing something here, why does this filter not match on all nodes?